# Selma in TSD
<h1 align="center">
  <br>
  <a href="https://github.com/elixir-no-nels/Selma"><img src="https://raw.githubusercontent.com/elixir-no-nels/Selma/master/.selma.svg?sanitize=true" alt="Selma" width="300"></a>
</h1>

## Description
Selma is a germline variant calling workflow developed at the University of Bergen. The workflow itself is based on Snakemake and all dependencies are handled by using Docker and Singularity container technology. The workflow has been tested on Colossus on TSD. Selma is named after the mythical Norwegian sea serpent that supposedly lives in Lake Seljord.

## Homepage
[https://github.com/elixir-no-nels/Selma](https://github.com/elixir-no-nels/Selma)

## Container version/release
Selma container version: 19.11

## Other relevant versions
[Singularity](https://github.com/elixir-no-nels/Selma/blob/master/docs/developer-instructions.md#cloning-this-repository-and-building-the-singularity-image)  
Docker: 19.03.2+  
[https://github.com/elixir-no-nels/Selma#tools](https://github.com/elixir-no-nels/Selma#tools)  

## Installation instructions
1. [TSD instructions](https://github.com/elixir-no-nels/Selma/blob/master/docs/TSD-instructions.md#installation)  
2. [Developer instructions](https://github.com/elixir-no-nels/Selma/blob/master/docs/developer-instructions.md)  

## Usage instructions
[Usage instructoins](https://github.com/elixir-no-nels/Selma/blob/master/docs/TSD-instructions.md#quickstart)

## Hardware/resource requirements
The workflow runs well with 16 cores and 3750MB of RAM per core.  
The staging disk requires roughly 100GB of temporary space to complete all file transfers.  
A typical execution with a total of 50GB input files takes 17-18 hours, the default time requested by SLURM is set to 25 hours and should be a decent margin.  
The user does not need to change the settings if the settings described above are desired. Adding more cores will shave some time off of the total execution time, but it does not scale linearly so it would be an "expensive" improvement if it is not critical to finish the workflow as soon as possible.  

## How the container was tested and results
### Test data
The provided toy fastq files were produced by running `head -n 40000 input_R1.fastq > output_R1.fastq` and `head -n 40000 input_R2.fastq > output_R2.fastq` on a larger pair of fastq files and are found in `Selma/workspace/fastq/`.  
The workflow will run all but the last step due to a limitation in that tool.  
There are production scale test files available on TSD as well. These files were generated by sequencing a Coriell cell line at the core facility at Haukeland hospital. The files are located in `/tsd/shared/bioinformatics/Test_Data/DNA_Fastqs/`

### Execution details for the "real size" data test
The testing instructions do not differ from the normal running instructions, refer to the [documentation](https://github.com/elixir-no-nels/Selma/blob/master/docs/TSD-instructions.md#quickstart) for complete instructions. The suggested testing files are either the toy files in `Selma/workspace/fastq` or in `/tsd/shared/bioinformatics/Test_Data/DNA_Fastqs/` on TSD.  

### Results for the "real size" dataset
A test with the provided fastq files should take about 10 minutes on a 16 core and 16GB machine, the disk should have 10GB free space to fit the reference files and output files. For production use you should consider at least a 16 core 16GB+ server with 500GB intermediary storage. The final output files should be less than 100GB per sample. The CPU usage will go up to 100% during certain parts of the workflow, but will remain low during other parts of the workflow.

### Support
Contact [Oskar Vidarsson](mailto:oskar.vidarsson@uib.no), [David Dolan](mailto:david.dolan@uib.no) or [Kjell Petersen](mailto:kjell.petersen@uib.no) for support enquiries. 
